src/kfactory/kcell.py

```diff
@@ -1404,6 +1404,9 @@ def to_yaml(cls, representer, node):  # type: ignore
         if shapes:
             d["shapes"] = shapes
         if len(node.settings) > 0:
+            for key, value in node.settings.items():
```

Comment:
Why is this necessary? When does it make sense to have a callable setting?

Reply:
See for instance the mzi pcell, where some arguments are functions to other pcells. Perhaps the arguments should always be kcells?